src/murmur/ServerDB.cpp

```diff
@@ -147,6 +147,44 @@ ServerDB::ServerDB() {
 		qFatal("ServerDB: Failed initialization: %s",qPrintable(e.text()));
 	}
 
+	// Use SQLite in WAL mode if possible.
+	if (Meta::mp.qsDBDriver == "QSQLITE") {
+		if (Meta::mp.iSQLiteWAL > 0 && Meta::mp.iSQLiteWAL <= 2) {
+			QSqlQuery query;
+
+			bool ok = false;
+			QString version;
+			SQLDO("SELECT sqlite_version();");
+			if (query.next()) {
+				version = query.value(0).toString();
+			}
+
+			QStringList splitVersion = version.split(QLatin1String("."));
+			int major = 0, minor = 0;
+			if (splitVersion.count() >= 3) {
+				major = splitVersion.at(0).toInt();
+				minor = splitVersion.at(1).toInt();
+			}
+			if (major >= 3 || (major == 3 && minor >= 7)) {
+				ok = true;
+			}
+
+			if (ok) {
+				SQLDO("PRAGMA journal_mode=WAL;");
+				if (Meta::mp.iSQLiteWAL == 1) {
+					SQLDO("PRAGMA synchronous=NORMAL;");
+				} else if (Meta::mp.iSQLiteWAL == 2) {
+					SQLDO("PRAGMA synchronous=FULL;");
+				}
+				qWarning("ServerDB: Configured SQLite for journal_mode=WAL, synchronous=NORMAL");
+			} else {
+				qWarning("ServerDB: Unable to use SQLite in WAL mode due to incompatible SQLite version %i.%i", major, minor);
```

Comment:
This will log version 0.0 if something with that went wrong. If both are 0, maybe log a warning on determining the version instead?